src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -807,69 +776,94 @@ pub fn expect_workunit_store_handle() -> WorkunitStoreHandle {
   get_workunit_store_handle().expect("A WorkunitStore has not been set for this thread.")
 }
 
-pub async fn with_workunit<F, M>(
+///
+/// NB: Public for macro usage: use the `in_workunit!` macro.
+///
+pub fn _start_workunit(
```

Comment:
I'm not sure how best to factor the macro with its two arms. For example, perhaps this function should no longer exist and we should inline. We could have one arm call the other for DRY.

Reply:
I think that it's ok for now. If it grows at all can DRY it up.